test/Yoti.Auth.Tests/ProfileTests.cs

```diff
@@ -10,16 +10,16 @@ namespace Yoti.Auth.Tests
     public class ProfileTests
     {
         [TestMethod]
-        public void Profile_GetAttribute()
+        public void Profile_GetAttribute_String()
         {
             var attributeValue = new YotiAttributeValue(TypeEnum.Text, Encoding.UTF8.GetBytes("value"));
-            var initialAttribute = new YotiAttribute<string>(Constants.UserProfile.PhoneNumberAttribute, attributeValue);
+            var initialAttribute = new YotiAttribute<string>(Constants.ApplicationProfile.ApplicationNameAttribute, attributeValue);
 
             ApplicationProfile applicationProfile = TestTools.Profile.CreateApplicationProfileWithSingleAttribute(initialAttribute);
```

Comment:
It's not immediately clear which class you are actually testing.
This test is for the base object, correct?  Or is it really for ApplicationProfile?

Reply:
Yes this file is for the base object. I guess it would be clearer if I split it up into `ApplicationProfileTests` and `UserProfileTests`, would you agree?

Reply:
Yes, if you are testing two different objects then please have two different tests.  

If you're only testing _ApplicationProfile_ then just call it _ApplicationProfileTest_.

If you're deliberately testing the base class, and you don't really care about _UserProfile_ or _ApplicationProfile_ here, then one approach is to create a simple subclass within your test (say, _TestProfile_) and just test that.